enhancements/windows-containers/windows-node-egress-proxy.md

```diff
@@ -0,0 +1,351 @@
+
+### Upgrade / Downgrade Strategy
+
+The relevant upgrade path is from WMCO 8.y.z in OCP 4.13 to WMCO 9.y.z in OCP 4.14. There will be no changes to the
+current WMCO upgrade strategy. Once customers are on WMCO 9.0.0, they can configure a cluster-wide proxy and the Windows
+nodes will be automatically updated by the operator to use the `Proxy` settings for egress traffic.
+
+When deconfiguring Windows instances, proxy settings will be cleared from the node. This involves undoing some node
+config steps i.e. unsetting proxy variables and deleting additional certificates from the machine's local trust store.
+This scenario will occur when upgrading both BYOH and Machine-backed Windows nodes.
+
+Downgrades are generally [not supported by OLM](https://github.com/operator-framework/operator-lifecycle-manager/issues/1177),
+which manages WMCO. In case of breaking changes, please see the
+[WMCO Upgrades](https://github.com/openshift/enhancements/blob/master/enhancements/windows-containers/windows-machine-config-operator-upgrades.md#risks-and-mitigations)
+enhancement document for guidance.
+
+### Version Skew Strategy
+
+N/A. There will be no version skew since this work is all within 1 product sub-component (WMCO). The 8.y.z version of the
+official Red Hat operator will not have cluster-wide egress proxy support for Windows enabled. Then, when customers move
+to WMCO 9.0.0, the proxy support will be available.
+
+## Implementation History
+
+The implementation history can be tracked by following the associated work items in Jira and source code improvements in
+the WMCO Github repo.
+
+## Alternatives & Justification
+
+### Design
+
+* Another possible way for WMCO to retrieve the proxy variables is to read them from its own pod spec, as these env
+  vars are injected and updated by OLM. The difficulty of this approach comes from figuring out when we need to update
+  node's env vars. Ideally such reconfiguring happens only when the values change in the pod spec, but how can we detect
+  if the proxy env vars changed or the operator just restarted for some other reason? We want to avoid kicking off
+  reconfigurations of all nodes every time the operator restarts.
+* In order to avoid a system reboot after setting node environment variables, we can reconcile services by setting their process-level
+  environment variables, and then restarting the services. These changes are ephemeral though, if the instance reboots. Either:
+  * For each service
+    ```powershell
+    [string[]] $envVars = @("HTTP_PROXY=", "NO_PROXY=123.example.com,10.88.0.0/16")
+    Set-ItemProperty HKLM:SYSTEM\CurrentControlSet\Services\<$SERVICE_NAME> -Name Environment -Value $envVars
+    Restart-Service <$SERVICE_NAME>
+    ```
+  * For the Powershell session
+    ```powershell
+    $Env:HTTP_PROXY = "http://<username>:<pswd>@<ip>:<port>"
+    $Env:NO_PROXY = "123.example.com,10.88.0.0/16"
+    # then restart all WMCO-managed services
+    ```
```

Comment:
Not rebooting is ideal. Should the variables be added to services-configmap? Or in separate secret which can be accessed by WICD? This seems appropriate as it directly relates to service configuration.
This way they will always be used, reboot or not.